sdk/core/Azure.Core/src/Shared/BearerTokenChallengeAuthenticationPolicy.cs

```diff
@@ -91,18 +80,18 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
             TokenRequestContext context;
 
             // If the message already has a challenge response due to a sub-class pre-processing the request, get the context from the challenge.
-            if (message.HasResponse && message.Response.Status == (int)HttpStatusCode.Unauthorized && message.Response.Headers.Contains(ChallengeHeader))
+            if (message.HasResponse && message.Response.Status == (int)HttpStatusCode.Unauthorized && message.Response.Headers.Contains(HttpHeader.Names.WWWAuthenticate))
             {
                 if (!TryGetTokenRequestContextFromChallenge(message, out context))
                 {
                     // We were unsuccessful in handling the challenge, so bail out now.
                     return;
                 }
-                _scopes = context.Scopes;
+                Scopes = context.Scopes;
```

Comment:
I wonder if we should cache the entire `TokenReqeustContext` rather than just the scopes. In the future this will likely also contain authority info which will need to be cached as well. This has the side effect of caching the claims challenge as well, but I don't think that will be an issue, but we'll need to test to make sure this isn't the case.

Reply:
Interesting idea. I'll look into it.

Reply:
After talking about it a bit offline, I'm going to punt on that change for now, since I think we'd never want to cache the ParentRequestId of the `TokenRequestContext`. We could just add authority to the policy when it's added to the `TokenRequestContext`.